packages/auth/src/signature/middleware.ts

```diff
@@ -277,3 +278,24 @@ export async function tokenHttpsigMiddleware(
   await verifySigFromBoundKey(grant, ctx)
   await next()
 }
+
+type Middleware = (ctx: AppContext, next: () => Promise<any>) => Promise<void>
+
+export function bypassOrCallHttpsigMiddleware(
+  bypassSignatureValidation: boolean,
+  middleware: Middleware
+): Middleware {
+  if (bypassSignatureValidation) {
+    return async (ctx: AppContext, next) => {
+      ctx.logger.info('Skipping httpsig validation')
+
+      ctx.clientKeyId = getSigInputKeyId(
+        ctx.headers['signature-input'] as string
+      ) //TODO: remove once workaround is found for clientKeyId definition
```

Comment:
Makes it possible to provide proper keyId via:

`Signature-Input: keyid="{{keyId}}"`

Reply:
Should this call `clientService.getKey`?
Otherwise, I think this will allow any `keyId`, which would cause token introspection to fail (if it isn't also bypassed)
https://github.com/interledger/rafiki/blob/48cdfa261ce6857810cd1e5a080c411788db5da2/packages/auth/src/accessToken/service.ts#L116-L119